### PR TITLE
[SYCL] Make SYCL geometrical functions overloads instead of templates 

### PR DESCRIPTION
**Pull request by KornevNikita**
It was decided to change all built-in functions from templates to overloads. This patch changes non-marray part of "4.17.8. Geometric functions".

Marray part: https://github.com/intel/llvm/pull/10048

Spec: https://github.com/KhronosGroup/SYCL-Docs/pull/428

### CODE REVIEW

File: sycl/include/sycl/builtins.hpp [hunk 1]
```diff
@@ -48,6 +48,287 @@ template <class T, int N> marray<T, N> to_marray(vec<T, N> x) {
 namespace __sycl_std = __host_std;
 #endif
 
+#define __SYCL_DEF_BUILTIN_VEC(TYPE)                                           \
```
Comment (KornevNikita) [inline sycl/include/sycl/builtins.hpp:51]: The major part of these defines will be added with https://github.com/intel/llvm/pull/9753, here we have them just to have compileable tests.

**Review by dm-vodopyanov — APPROVED** [review]

File: sycl/include/sycl/builtins.hpp [hunk 2]
```diff
@@ -60,6 +60,10 @@ namespace __sycl_std = __host_std;
   __SYCL_BUILTIN_DEF(TYPE##3)                                                  \
   __SYCL_BUILTIN_DEF(TYPE##4)
 
+#define __SYCL_DEF_BUILTIN_GEOCROSSVEC(TYPE)                                   \
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/builtins.hpp:63]: Please add `#undef` for all new macros which was introduced in this patch

**Review by dm-vodopyanov — CHANGES_REQUESTED** [review]
Sorry, revoked "Approved"

File: sycl/include/sycl/builtins.hpp (hunk 2 above)
Comment (KornevNikita) [reply sycl/include/sycl/builtins.hpp:63]: Oh yes, forgot about it, thanks! [c72b5b2](https://github.com/intel/llvm/pull/10014/commits/c72b5b29146242886d34375e6f060b57443ad23e)

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by KornevNikita** [comment]
Converting to draft to check something.

**Pull request merged**